coderd/httpmw/apikey.go

```diff
@@ -812,3 +812,24 @@ func CustomRedirectToLogin(rw http.ResponseWriter, r *http.Request, redirect str
 
 	http.Redirect(rw, r, u.String(), code)
 }
+
+// ExtractAPIKeyFromHeader extracts an authorization token from HTTP headers.
+// It checks the Authorization header (Bearer token) and X-Api-Key header,
+// which represent the different ways clients authenticate against AI providers.
+// If neither are present, an empty string is returned.
+//
+// This differs from [APITokenFromRequest] in that it only checks headers commonly
+// used by AI providers, not Coder-specific authentication methods like cookies
+// or the Coder-Session-Token header.
+func ExtractAPIKeyFromHeader(header http.Header) string {
```

Comment:
If we move this here, we have to name this very carefully because it's only used in specific situations (i.e. aibridge).

I'd suggest moving this into a non-enterprise aibridge package.